### DISCUSSION TITLE
Allow suggest text input as dynamic in useTranslations

### DISCUSSION THREAD

**Discussion by heliocastro** [amannn/next-intl#669] [Q&A]
### Is your feature request related to a problem? Please describe.

Using typescript eslint 6, dynamic text based on variables issues a TS 2345 error even though the code has been correctly executed. 

Example:
![image](https://github.com/amannn/next-intl/assets/418176/95175344-3b09-4636-9733-c611949c9ad2)

To override the solution, ignoring the error can be done this way:
```typescript
{
      // @ts-expect-error: TS2345 invalidate translation even if is valid under
       t(tab.name)
}
```

System used: MacOS Sonoma 14.1
NodeJS 20.7.0
Next 14.0.3
Next-Intl 3.1.4

### Describe the solution you'd like

Possibility of useTranslations t function with a parameter that accepts a string as a valid entry without the need to validate existence in the message list.

### Describe alternatives you've considered

No alternative is considered as using ts error ignore the current method.

**Comment by amannn** [comment]
You can certainly use variables in the `t` function but TypeScript rightfully requires that you're using a string union.

**Examples:**
```tsx
const k1: string = 'maybe-valid-key';
t(k1); // Error

const k2: 'valid-key-1' | 'valid-key-2' = 'valid-key-1';
t(k2); // Ok
```

I'll move this to a discussion since this is a usage question.

**Comment by heliocastro** [comment]
Thank you for quick answer, appreciated.

Yes, that would be correct if i know ahead the messages. :-)
I realize that our dynamic components are wrong, based on assumption to translate inside the component.
We should submit the already translated message and this problem would go away.

Before was not possible due to lack of non client translation capability, but now changed.

Still, the first t(k1) case still could be valid if you want concentrate the translations on the component, but based on your, input, we need rethink our components.
The idea in general was retrieve possible list of messages that i don't know from a rest service and create the component entries with this, that could or not be translated.
